### PR TITLE
Main View Toggle

### PR DESCRIPTION
**Pull request by jameskerr**
Part of #1325 

The PR contains a few things.

1. A button that toggles the "mainView" from "summary to "search"
2. An update to the layout reducer to save the "mainView"
3. A new Feature reducer that enables us to turn a feature on and off.

Up close of the button
Mh8IdSVGun

Up close of the feature flag
ZlumUnmDx0

The feature will be off by default so that we can slowly fill out the security dashboard in multiple PRs, while always being able to release.

### CODE REVIEW

**Review by mason-fish — APPROVED** [review]
Looks good! Just 2 comments. I like that `switch-button` too!

File: app/core/hooks/useLayoutMount.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {useLayoutEffect, useRef} from "react"
+
+export default function useLayoutMount(fn: () => void) {
```
[the remark this answers is not in the document]
Comment (jameskerr) [reply app/core/hooks/useLayoutMount.ts:3]: Yes, you're right. I'll change to that.

File: app/core/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    fill: ${(p) => p.fill || css`var(--slate)`};
+    stroke: ${(p) => p.stroke || "inherit"};
+  }
+`
+
+export default function Icon(props: Props) {
+  const SVG = icons[props.name]
+  return (
+    <Wrap
+      className={classNames(props.className, `${props.name}-icon`)}
```
[the remark this answers is not in the document]
Comment (jameskerr) [reply app/core/Icon.tsx:32]: Sounds good. I did a grep and we have around 40 icons using the "old" pattern. For ease of a transition, I thought of keeping the classname. But maybe it wouldn't be that hard to convert them to Styled Components. Here are all the svgs in the repo:

[the remark this answers is not in the document]
Comment (jameskerr) [reply app/core/Icon.tsx:32]:
```
src/js/components/icons/carrot-down.svg
src/js/components/icons/back-arrow.svg
src/js/components/icons/icon-asc.svg
src/js/components/icons/warning-sm.svg
src/js/components/icons/x-md.svg
src/js/components/icons/star-sm.svg
src/js/components/icons/forward-arrow.svg
src/js/components/icons/chevron-left-md.svg
src/js/components/icons/circle-x-md.svg
src/js/components/icons/caret-bottom-sm.svg
src/js/components/icons/icon-desc.svg
src/js/components/icons/pin-md.svg
src/js/components/icons/warning-md.svg
src/js/components/icons/carrot-up.svg
src/js/icons/ramp-right.svg
src/js/icons/clock-icon.svg
src/js/icons/sharkfin.svg
src/js/icons/file-fill.svg
src/js/icons/logo-type.svg
src/js/icons/expand-window-icon.svg
src/js/icons/brim-text.svg
src/js/icons/trash-bin.svg
src/js/icons/ramp-left.svg
src/js/icons/octocat.svg
src/js/icons/columns-icon.svg
src/js/icons/folder.svg
src/static/icons/subspace.svg
src/static/icons/export.svg
src/static/icons/subspace-border.svg
src/static/icons/zeek-file.svg
src/static/icons/three-dots.svg
src/static/icons/pin.svg
src/static/icons/book.svg
src/static/icons/file-border.svg
src/static/icons/circle-close.svg
src/static/icons/file-filled.svg
src/static/icons/star-no-fill.svg
src/static/icons/star.svg
src/static/icons/pcap-file.svg
src/static/icons/view.svg
src/static/icons/magnifying-glass.svg
src/static/icons/archive-border.svg
src/static/icons/pin-border.svg
```

**Pull request merged**